### PR TITLE
Support test, build, and push for CentOS Stream 9

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support testing, building, and pushing images in CentOS Stream 9.

### CODE REVIEW

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
`pipenv-test-app` tests failed:

```bash
[INFO] Waiting for application container to start  ...

ERROR: don't know how to run your application.
Please set either APP_MODULE, APP_FILE or APP_SCRIPT environment variables, or create a file 'app.py' to launch your application.
```

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by phracek** [comment]
[test]

**Comment by frenzymadness** [comment]
Let me know if there is anything I should investigate.

**Comment by phracek** [comment]
[test-all]

**Comment by frenzymadness** [comment]
This seems to be a real problem. I need some time to debug it.

**Comment by frenzymadness** [comment]
The problem is in Dockerfile because it uses macros for centos but I think that Centos stream is now closer to Fedora than Centos Linux was. The problem is that centos and centos stream share the same id in distgen which is not correct I believe. I'm gonna open an issue in distgen.

**Comment by frenzymadness** [comment]
https://github.com/devexp-db/distgen/issues/120

**Comment by frenzymadness** [comment]
@phracek right now, as we have the new release of distgen, we should create a new symlink `centos-stream` (the distgen id) pointing to `fedora` in `src` folder - the same way as `rhel` points to `centos`. We might need to create a separated folder with dedicated macros later but I think that fedora macros should work for centos stream as well.

**Comment by frenzymadness** [comment]
Do you want to do it or should I take over this PR?

**Comment by phracek** [comment]
The distgen package was updated by COPR build https://copr.fedorainfracloud.org/coprs/praiskup/distgen/build/3825089/
Let's see what happens.

[test]

**Comment by frenzymadness** [comment]
> The distgen package was updated by COPR build https://copr.fedorainfracloud.org/coprs/praiskup/distgen/build/3825089/ Let's see what happens.

There is no symlink I've mentioned in:

> @phracek right now, as we have the new release of distgen, we should create a new symlink `centos-stream` (the distgen id) pointing to `fedora` in `src` folder - the same way as `rhel` points to `centos`. We might need to create a separated folder with dedicated macros later but I think that fedora macros should work for centos stream as well.

The result on C9S is: `Error: This repository does not have any builds yet so you cannot enable it now.` Not sure why because I see that the latest build is available for all platforms including EPEL 9. Will check.

**Comment by frenzymadness** [comment]
> The result on C9S is: `Error: This repository does not have any builds yet so you cannot enable it now.` Not sure why because I see that the latest build is available for all platforms including EPEL 9. Will check.

Now I see it. There is a specific chroot for centos-stream-9 which has to be enabled if we want to install distgen directly on C9S from the COPR repo. Cc: @praiskup

**Comment by phracek** [comment]
> b

This is already fixed. https://github.com/sclorg/sclorg-testing-farm/commit/622ef39bfe0aeb7ebdbeb692199d199ac49355fa

**Comment by phracek** [comment]
[test]

**Comment by frenzymadness** [comment]
[test]

**Comment by frenzymadness** [comment]
@phracek With the latest common scripts, there is this error:
```
-> Image for version 3.8 not built, skipping tag.
-> Tagging image 'ffbf376abc1e5fac74f2281922b5b39953a73160dfec3b29b5816d745ae42b9a' as 'quay.io//python3:0' and 'quay.io//python3:latest' and 'quay.io//python3:20220329-f4c94ca'
-> Tagging image 'ffbf376abc1e5fac74f2281922b5b39953a73160dfec3b29b5816d745ae42b9a' as 'quay.io//python3:c9s'
Error: normalizing name "quay.io//python3:c9s": error normalizing name "quay.io//python3:c9s": invalid reference format
make: *** [common/common.mk:120: tag] Error 125
```
It seems to be caused by this commit: https://github.com/sclorg/container-common-scripts/commit/2a500813557f7c9c7d6d6654c8c5bf6aabf608b2 
I'm gonna downgrade the submodule and try it again.

**Comment by frenzymadness** [comment]
[test]

**Comment by frenzymadness** [comment]
[test]

File: 3.9/Dockerfile.c9s [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+on most platforms."
+
+LABEL summary="$SUMMARY" \
+      description="$DESCRIPTION" \
+      io.k8s.description="$DESCRIPTION" \
+      io.k8s.display-name="Python 3.9" \
+      io.openshift.expose-services="8080:http" \
+      io.openshift.tags="builder,python,python39,python-39,rh-python39" \
+      com.redhat.component="$NAME" \
+      name="$FGC/$NAME" \
```
Comment (phracek) [inline 3.9/Dockerfile.c9s:36]: This has to be in format `sclorg/python-39-c9s`

**Comment by frenzymadness** [comment]
[test]

**Comment by frenzymadness** [comment]
The last run of tests for C9S was green so I did a rebase and cleaned the commits. Let's see now, [test]

**Review by phracek — COMMENTED** [review]
LGTM. Thanks for the PR. C9S are green.

**Comment by phracek** [comment]
[test-openshift]

**Comment by phracek** [comment]
[test]

**Comment by frenzymadness** [comment]
It seems ready to be merged!

**Comment by phracek** [comment]
Let's get merge it.

**Pull request merged**